crates/nu-command/src/filesystem/glob.rs

```diff
@@ -186,6 +185,21 @@ impl Command for Glob {
             }
         };
 
+        let path = current_dir(engine_state, stack)?;
+        let path = match nu_path::canonicalize_with(prefix, path) {
+            Ok(path) => path,
+            Err(e) if e.to_string().contains("os error 2") => std::path::PathBuf::new(),
```

Comment:
Remind me, what exactly is `os error 2` and why are we eating it?

Reply:
Means "file or directory not found".  And it's a fair question! Happens if user specifies a directory prefix before the wildcard where the directory doesn't exist. We set empty path to `Glob::walk()` and it returns an empty list.
I added a comment because I couldn't immediately figure it out even though I wrote this code only a day ago!